src/MacroTools/Commands/Share.cs

```diff
@@ -23,20 +24,43 @@ public sealed class Share : Command
     /// <inheritdoc />
     public override string Execute(player cheater, params string[] parameters)
     {
+      var cheaterTeam = cheater.GetTeam(); // Declare cheaterTeam once at the beginning
+
+      // Check if the first parameter is 'all', regardless of any additional parameters
+      if (parameters.Length >= 1 && parameters[0].ToLower() == "all")
+      {
+        var factions = FactionManager.GetAllFactions();
+
+        foreach (var faction in factions)
+        {
+          if (faction.Player != null && faction.Player.GetTeam() == cheaterTeam)
+          {
+            SetPlayerAlliance(cheater, faction.Player, ALLIANCE_SHARED_CONTROL, true);
+          }
+        }
+
+        return $"Shared control with all factions on your team.";
+      }
+
+      // If not 'all', then proceed with previous share logic from Yak's code
+      if (parameters.Length == 0 || parameters[0].ToLower() == "")
```

Comment:
You can get rid of this whole if statement. The intention is that `CommandManager` handles making sure the user provides the minimum set of parameters.